core/iwasm/compilation/aot_emit_gc.c

```diff
@@ -1446,21 +1442,22 @@ aot_compile_op_array_copy(AOTCompContext *comp_ctx, AOTFuncContext *func_ctx,
         goto fail;
     }
 
-    if (!aot_emit_exception(comp_ctx, func_ctx, EXCE_NULL_GC_REF, true, cmp[0],
-                            check_objs_succ))
+    if (!aot_emit_exception(comp_ctx, func_ctx, EXCE_NULL_ARRAY_OBJ, true,
+                            cmp[0], check_objs_succ))
         goto fail;
 
     /* Create if block */
     ADD_BASIC_BLOCK(len_gt_zero, "len_gt_zero");
     MOVE_BLOCK_AFTER_CURR(len_gt_zero);
     /* Create inner else block */
     ADD_BASIC_BLOCK(inner_else, "inner_else");
-    MOVE_BLOCK_AFTER(inner_else, len_le_zero);
 
     /* Create else(end) block */
     ADD_BASIC_BLOCK(len_le_zero, "len_le_zero");
     MOVE_BLOCK_AFTER_CURR(len_le_zero);
 
+    MOVE_BLOCK_AFTER(inner_else, len_le_zero);
```

Comment:
Had better move `ADD_BASIC_BLOCK` above to here, and change `len_le_zero` to `len_gt_zero` (inner_else is inside len_gt_zero)?
```C
    /* Create inner else block */
    ADD_BASIC_BLOCK(inner_else, "inner_else");
    MOVE_BLOCK_AFTER(inner_else, len_gt_zero);
```